File: student-service/src/main/java/kz/baltabayev/studentservice/model/service/FacultyService.java
package kz.baltabayev.studentservice.model.service;

import kz.baltabayev.studentservice.model.entity.Faculty;

import java.util.List;

public interface FacultyService {

    List<Faculty> getAll();

    Faculty getById(Long id);

    Faculty create(Faculty faculty);

    Faculty update(Faculty faculty);

    void deleteFacultyById(Long id);
}

File: faculty-department-service/src/main/kotlin/kz/baltabayev/facultydepartmentservice/service/FacultyService.kt
package kz.baltabayev.facultydepartmentservice.service

import kz.baltabayev.facultydepartmentservice.model.dto.FacultyDto
import kz.baltabayev.facultydepartmentservice.model.entity.Faculty

interface FacultyService {

    fun save(facultyDto: FacultyDto)

    fun update(facultyDto: FacultyDto, id: Long)

    fun delete(id: Long)

    fun findById(id: Long): Faculty
}